src/include/crowbar/complex.rb

```diff
@@ -695,12 +617,195 @@ def StoreRepoURL(key, event)
     def HandleRepoURL(key, event)
       # store the value on exiting
       if event["ID"] == :next || event["EventReason"] == "ValueChanged"
+        @current_repo = UI.QueryWidget(Id("repos_table"), :Value) || ""
         StoreRepoURL(key, event)
         InitReposTable("repos_table")
       end
       nil
     end
 
+    def InitLocationURL(id)
+      UI.ChangeWidget(Id(id), :Value, @remote_server_url)
+      nil
+    end
+
+    def StoreLocationURL(key, event)
+      @remote_server_url = UI.QueryWidget(Id(key), :Value)
+      nil
+    end
+
+    def HandleLocationURL(key, event)
+
+      StoreLocationURL(key, event)
+
+      return nil if @remote_server_url.empty? || @repos_location == "custom"
+
+      @repos.each do |prod_name, platform|
+        distro = prod_name == "suse-11.3" ? "sles11-sp3-x86_64" : "sles12-x86_64"
+        platform.each do |repo_name, r|
+          url = "#{@remote_server_url}/repo/$RCE/#{repo_name}/x86_64/"
+          if @repos_location == "sm"
+            if ["SLE-Cloud","SLE-Cloud-PTF"].include? repo_name
+              # some repos cannot be at SM server
```

Comment:
I think some reference why would be nice? why suse manager cannot stage cloud repos?